autoemulate/experimental/sensitivity_analysis.py

```diff
@@ -0,0 +1,264 @@
+        """
+        Get the output names from the problem definition or generate default names.
+        """
+        # check if output_names is given
+        if "output_names" not in self.problem:
+            output_names = [f"y{i + 1}" for i in range(num_outputs)]
+        elif isinstance(self.problem["output_names"], list):
+            output_names = self.problem["output_names"]
+        else:
+            msg = "'output_names' must be a list of strings."
+            raise ValueError(msg)
+
+        return output_names
+
+    def run(
+        self,
+        method: str = "sobol",
+        N: int = 1024,
+        conf_level: float = 0.95,
+    ) -> pd.DataFrame | None:
+        """
+        Perform global sensitivity analysis on a fitted emulator.
+
+        Parameters
+        ----------
+        method: str
+            The sensitivity analysis method to perform, one of ["sobol", "morris"].
+        N : int
+            Number of samples to generate for the analysis. Higher values give more
+            accurate results but increase computation time. Default is 1024.
+        conf_level : float
+            Confidence level (between 0 and 1) for calculating confidence intervals
+            of the Sobol sensitivity indices. Default is 0.95 (95% confidence). This
+            is not used in Morris sensitivity analysis.
+
+        Returns
+        -------
+        pandas.DataFrame
+            DataFrame with columns:
+                - 'parameter': Input parameter name
+                - 'output': Output variable name
+                - 'S1', 'S2', 'ST': First, second, and total order sensitivity indices
+                - 'S1_conf', 'S2_conf', 'ST_conf': Confidence intervals for each index
+
+        Notes
+        -----
+        The Sobol method requires N * (2D + 2) model evaluations, where D is the number
+        of input parameters. For example, with N=1024 and 5 parameters, this requires
+        12,288 evaluations. The Morris method requires far fewer computations.
+        """
+        if method not in ["sobol", "morris"]:
+            msg = f"Unknown method: {method}. Must be 'sobol' or 'morris'."
+            raise ValueError(msg)
+
+        param_samples = self._sample(method, N)
+        y = self._predict(param_samples)
+        output_names = self._get_output_names(y.shape[1])
+
+        results = {}
+        for i, name in enumerate(output_names):
+            if method == "sobol":
+                Si = sobol_analyze(self.problem, y[:, i], conf_level=conf_level)
+            elif method == "morris":
+                Si = morris_analyze(self.problem, param_samples, y[:, i])
+            results[name] = Si  # type: ignore PGH003
+
+        if method == "sobol":
+            return _sobol_results_to_df(results)
+        if method == "morris":
+            return _morris_results_to_df(results, self.problem)
+
+        # this should never happen, just keeping type checker happy
+        return None
```

Comment:
I think given the check on L200 we might be able to refactor this to (and remove the `None` from the return type):
```suggestion
        return _morris_results_to_df(results, self.problem)
```